### PR TITLE
respect the CLOUDSDK_CONFIG

### PR DESCRIPTION
**Pull request by lesv**
Make DefaultCredentialsProvider respect the CLOUDSDK_CONFIG environment
variable.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # Maven
-target
+target/
```
Comment (anthmgoogle) [inline .gitignore]: Put the .gitignore changes into a separate PR/CL?

**Comment by anthmgoogle** [comment]
Suggest separating out the .gitignore changes. Otherwise LGTM. Since you are admin, merge when you are ready.

**Comment by ejona86** [comment]
@lesv LGTM

**Comment by lesv** [comment]
I'm going to go w/ merged PR, but I do understand that for the future, I'll do tiny changes.  (My git foo isn't  nimble enough to backout just one file.

**Pull request merged**